Kickstarter-iOS/Views/Controllers/LiveStreamDiscoveryViewController.swift

```diff
@@ -0,0 +1,81 @@
+import KsApi
+import Library
+import LiveStream
+import Prelude
+
+internal final class LiveStreamDiscoveryViewController: UITableViewController {
+  private let dataSource = LiveStreamDiscoveryDataSource()
+  private let viewModel: LiveStreamDiscoveryViewModelType = LiveStreamDiscoveryViewModel()
+
+  internal override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.tableView.dataSource = self.dataSource
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  internal func isActive(_ active: Bool) {
+    self.viewModel.inputs.isActive(active)
+  }
+
+  internal override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> baseTableControllerStyle(estimatedRowHeight: 200.0)
+  }
+
+  internal override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.loadDataSource
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.dataSource.load(liveStreams: $0)
+        self?.tableView.reloadData()
+    }
+
+    self.viewModel.outputs.goToLiveStreamCountdown
+      .observeForControllerAction()
+      .observeValues { [weak self] project, stream, event in
+        self?.goToLiveStreamCountdown(project: project, stream: stream, liveStreamEvent: event)
+    }
+
+    self.viewModel.outputs.goToLiveStreamContainer
+      .observeForControllerAction()
+      .observeValues { [weak self] project, stream, event in
+        self?.goToLiveStreamContainer(project: project, stream: stream, liveStreamEvent: event)
+    }
+  }
+
+  private func goToLiveStreamContainer(project: Project,
+                                       stream: Project.LiveStream,
+                                       liveStreamEvent: LiveStreamEvent) {
+    let vc = LiveStreamContainerViewController.configuredWith(project: project,
+                                                              liveStream: stream,
+                                                              event: liveStreamEvent)
+    let nav = UINavigationController.init(navigationBarClass: ClearNavigationBar.self, toolbarClass: nil)
+    nav.viewControllers = [vc]
+    DispatchQueue.main.async {
+      self.present(nav, animated: true, completion: nil)
+    }
```

Comment:
we were seeing some weird flickering of the screen when not using `DispatchQueue.main.async` :/ it seems to be entirely due to initializing the nav controller and _then_ setting it's controllers, as opposed to using the `UINavigationController(rootViewController:)` initializer.